lona/html/data_binding/inputs.py

```diff
@@ -38,13 +38,14 @@ def handle_change(self, input_event):
         # The solution for this problem is to don't send patches back to users
         # who issued them.
 
-        self.attributes.__setitem__(
-            'value',
-            input_event.data,
-            issuer=(input_event.connection, input_event.window_id),
-        )
+        with self.lock:
```

Comment:
Why is this necessary?  ``AttributeDict.__setitem__()`` pulls the lock for us.

Reply:
Lock may be dropped if we can drop we can drop `self.value = input_event.data` and use only `self.attributes.__setitem__`.
Otherwise there is a data race between them.

But you are right, that this change should be in a separate PR.